pkg/apis/autoscaling/v1/conversion.go

```diff
@@ -404,7 +404,10 @@ func Convert_v1_HorizontalPodAutoscaler_To_autoscaling_HorizontalPodAutoscaler(i
 	if currentConditionsEnc, hasCurrentConditions := out.Annotations[autoscaling.HorizontalPodAutoscalerConditionsAnnotation]; hasCurrentConditions {
 		var currentConditions []autoscalingv1.HorizontalPodAutoscalerCondition
 		if err := json.Unmarshal([]byte(currentConditionsEnc), &currentConditions); err != nil {
-			return err
+			//ignore unmarshalling errors because we persist invalid annotations into v1 objects
+			// Issue - https://github.com/kubernetes/kubernetes/issues/88738
+			delete(out.Annotations, autoscaling.HorizontalPodAutoscalerConditionsAnnotation)
```

Comment:
Not sure if this logic is ok. Could some of the content already be partially written to `currentConditions` above?

@liggitt Could you also take a look if the logic is fine?

Reply:
as long as we are not writing conditions back to `out.Status.Conditions` if/when there is unmarshalling errors, shouldn't that be enough ?